application-collabora-ui/src/main/resources/Collabora/Code/Main.xml

```diff
@@ -264,8 +282,33 @@ require(['jquery', 'xwiki-l10n!collabora-editor'], function($, l10n) {
       );
     } else if (msg.MessageId == 'Doc_ModifiedStatus') {
       isDocModified = msg.Values &amp;&amp; msg.Values.Modified;
+      if (isDocModified == false) {
+        const fileId = $("#collaboraServer").data('fileId');
+        const tokenUpdateURL = XWiki.contextPath + collaboraPath + encodeURIComponent(fileId) + '/token/extend';
+        $.ajax({
+          url: tokenUpdateURL,
+          type: 'PUT',
+          dataType: 'text',
+          success: function(resp, textStatus, jqXHR) {
+            if (jqXHR.status == 200) {
+              iframeWindow.postMessage(
+                JSON.stringify({ 'MessageId': 'Action_Save' }),
+                $("#collaboraServer").data('server')
```

Comment:
I'll add a comment here explaining why a new safe is needed (i.e. in case it's needed, we cannot renew the token before the Collabora safe, so we renew the token and then trigger a new safe), stating also if there is a problem in case the window is closed before the second safe is done